### PR TITLE
rename all references from multicloudhub to multiclusterhub

### PR DESCRIPTION
**Pull request by nathanweatherly**
https://app.zenhub.com/workspaces/engineering-backlog-do-not-delete-604fab62d4b98d00150a2854/issues/open-cluster-management/backlog/10938

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func main() {
 
 	ctx := context.TODO()
 	// Become the leader before proceeding
-	err = leader.Become(ctx, "multicloudhub-operator-lock")
+	err = leader.Become(ctx, "multiclusterhub-operator-lock")
```
Comment (JakobGray) [inline cmd/manager/main.go:104]: Changing the name of the lock might break consensus because they won't be sharing the same lock

Comment (nathanweatherly) [reply cmd/manager/main.go]: reverted

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**